### PR TITLE
Simplify external completer

### PR DESCRIPTION
**Pull request by JoaquinTrinanes**
This PR modernizes the external completer docs by using `match` instead of `{ ... } | get -i ... | default ...`.

It's:
- Simpler to read and understand
- A nice way to get folks reading the docs to learn about `match` (<3)
- A textbook example of when to use `match`

### CODE REVIEW

File: cookbook/external_completers.md [hunk 1]
```diff
@@ -68,18 +68,20 @@ Sometimes, a single external completer is not flexible enough. Luckily, as many
...
+        ls => $ls_completer
+        git => $git_completer
+        _ => $default_completer
+    } | do $in $spans
 }
 ```
 
 > **Note**
-> In the example above, `$spans.0` is the command being run at the time. The completer will try to `get` the desired completer in the record, and fallback to `$default_completer`.
+> In the example above, `$spans.0` is the command being run at the time. The completer will match the desired completer in the record, and fallback to `$default_completer`.
```
Comment (sholderbach) [inline cookbook/external_completers.md:81]:
```suggestion
> In the example above, `$spans.0` is the command being run at the time. The completer will match the desired completer, and fallback to `$default_completer`.
```

**Review by sholderbach — COMMENTED** [review]
Nice! It's a `match` ❤️

**Comment by rgwood** [comment]
Nice improvement, thanks!

File: cookbook/external_completers.md [hunk 2]
```diff
@@ -134,20 +136,27 @@ let carapace_completer = {|spans: list<string>|
...
+    let spans = if $expanded_alias != null {
+        $spans
+        | skip 1
+        | prepend ($expanded_alias | split row ' ')
+    } else {
+        $spans
+    }
 
-    {
+    match {
```
Comment (fdncred) [inline cookbook/external_completers.md:151]: What is `match` matching on here?

**Comment by fdncred** [comment]
@JoaquinTrinanes Can you please take a look at my comment above. I was trying out this cleanup and noticed that `match` isn't matching on anything.

File: cookbook/external_completers.md (hunk 2 above)
Comment (JoaquinTrinanes) [reply cookbook/external_completers.md:151]: Oops, that's a typo. It should be `match $spans.0` :)

I'll create a followup PR soon when I have time

**Pull request merged**